plugins/experimental/cachekey/cachekey.cc

```diff
@@ -285,7 +286,9 @@ CacheKey::appendPrefix(const String &prefix, Pattern &prefixCapture, Pattern &pr
     StringVector captures;
     if (prefixCapture.process(hostAndPort, captures)) {
       for (auto &capture : captures) {
-        append(capture);
+        if (!capture.empty()) {
+          append(capture);
+        }
```

Comment:
Skipping zero-length captures would change the behavior of the [capture definitions](https://docs.trafficserver.apache.org/en/latest/admin-guide/plugins/cachekey.en.html?#capture-definition) where the `regex` part contains multiple capture groups but no `replacement` part in which case all captures would be added as different elements in the cache key, i.e. `--capture-prefix=s([^.]*).(example).(net|com)`

The behavior of [capture definitions](https://docs.trafficserver.apache.org/en/latest/admin-guide/plugins/cachekey.en.html?#capture-definition) where the `replacement` part is defined would not change, i.e `--capture-prefix=/s([^.]*).(example).(net|com)/$1\/$2\/$3/`

This change is unrelated to what we are trying to solve with this patch and since we also plan to back-port it to 6.2.x decided to remove it (will reconsider adding it in 8.0.x).